### ISSUE TITLE
playground: export to JSON fails where cmd/cue does not

### ISSUE THREAD

**Issue by myitcv** [cue-lang/cue#1343]
### What version of CUE are you using (`cue version`)?

<pre>
$ cue version
v0.4.0 (per playground)
</pre>

### Does this issue reproduce with the latest release?

Yes

### What did you do?

Using JSON export:

```
#A: {
	num: number
}

#B: {
	ans: string
}

#C: {#A, #B}


one: #C & {
    num: 123
    ans: "hi"
}
```

gives:

```
failed to encode: #C: field not allowed: num (and 3 more errors)
```

e.g. https://cuelang.org/play/?id=BCTUQT1PYFC#cue@export@json

This works as expected with `cmd/cue`:

```
exec cue eval --out json x.cue

-- x.cue --
#A: {
	num: number
}

#B: {
	ans: string
}

#C: {#A, #B}

one: #C & {
	num: 123
	ans: "hi"
}
```

gives:

```
> exec cue eval --out json x.cue
[stdout]
{
    "one": {
        "num": 123,
        "ans": "hi"
    }
}
```

### What did you expect to see?

Success in the playground.

### What did you see instead?

The error above.

**Comment by antong** [comment]
Does the playground do an extra equivalent of `cue def input.cue` before exporting?  There seems to be many of these issues where cue def output is not equivalent to the input, and these trigger in the playground, e.g., https://github.com/cue-lang/cue/issues/1284#issuecomment-956157439 .

For this issue:
```
exec cue def x.cue

-- x.cue --
#A: {
        num: number
}

#B: {
        ans: string
}

#C: {#A, #B}
```

In the input `#C` embeds `#A` and `#B`, but not in the output (v0.4.0):
```
> exec cue def x.cue
[stdout]
#A: {
        num: number
}
#B: {
        ans: string
}
#C: #A & #B
```

**Comment by myitcv** [comment]
> Does the playground do an extra equivalent of `cue def input.cue` before exporting

@antong there are very likely a number of discrepancies between `cmd/cue` and the playground because they do not share code paths... and they should. #1150 broadly covers ensuring that the two are, modulo UX, equivalent for exactly this reason.

**Comment by mvdan** [comment]
This appears to have been fixed as of v0.4.3 as well as v0.5.0-rc.1 :) Closing.
